axelor-stock/src/main/java/com/axelor/apps/stock/service/TrackingNumberService.java

```diff
@@ -111,9 +113,18 @@ public TrackingNumber createTrackingNumber(Product product, Company company, Loc
           product.getCode());
     }
 
-    String seq = sequenceService.getSequenceNumber(trackingNumberConfiguration.getSequence());
-
-    trackingNumber.setTrackingNumberSeq(seq);
+    Sequence sequence = trackingNumberConfiguration.getSequence();
+    List<TrackingNumber> trackingNumbers =
+        trackingNumberRepo.all().filter("self.product = ?1", product).fetch();
```

Comment:
don't search trackingNumber just with product but search it with sequence number too, because its common that one product have many trackingNumber but its rare that its with same sequence.